File: new-algorithm/src/javaalgo/implementation/boj2753.java
package javaalgo.implementation;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class boj2753 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int year = Integer.parseInt(br.readLine());

        if (year % 4 == 0) {
            if (year % 400 == 0) System.out.println("1");
            else if (year % 100 == 0) System.out.println("0");
            else System.out.println("1");
        } else {
            System.out.println("0");
        }
    }
}

File: new-algorithm/src/kotlinalgo/implementation/boj2753.kt
package kotlinalgo.implementation

import java.io.BufferedReader
import java.io.InputStreamReader import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val year: Int = Integer.parseInt(readLine())

    println(when {
        year % 4 == 0 && (year % 400 == 0 || year % 100 != 0) -> "1"
        else -> "0"
    })

}